src/java/org/apache/cassandra/locator/DynamicEndpointSnitch.java

```diff
@@ -214,19 +600,27 @@ public String getDatacenter(InetAddressAndPort endpoint)
         {
             if (subsnitchScore > (sortedScoreIterator.next() * (1.0 + dynamicBadnessThreshold)))
             {
-                return sortedByProximityWithScore(address, replicas);
+                return sortedByProximityWithScore(address, replicas, aliasedScores);
             }
         }
 
         return replicas;
     }
 
     // Compare endpoints given an immutable snapshot of the scores
-    private int compareEndpoints(InetAddressAndPort target, Replica a1, Replica a2, Map<InetAddressAndPort, Double> scores)
+    public int compareEndpoints(InetAddressAndPort target, Replica a1, Replica a2)
```

Comment:
This comment seems wrong and the entire scenario is a little confusing?
Why can't this be removed?

Reply:
Not sure, I didn't write it just copied it over. I think the issue is that `IEndpointSnitch` requires `compareEndpoints(InetAddressAndPort, Replica, Replica)` to be defined so that `AbstractEndpointSnitch::sortedByProximity` works properly, but the only time that the DES would call it is if the DES is wrapping another DES (as `compareEndpoints(InetAddressAndPort, Replica, Replica, Map<InetAddressAndPort, Double>)` internally calls the `subsnitch.compareEndpoints` if it can't find a score.

Reply:
Ah I get it, it's to catch rewrapping I guess. Fine to leave it.